### DISCUSSION TITLE
Patch + Insert item (similar to Upsert)

### DISCUSSION THREAD

**Discussion by GopalanHarish** [Azure/azure-cosmos-dotnet-v3#2977] [Q&A]
Is there a way to create a document if `PatchItemAsync()` fails with 404 Not Found? I am looking for something similar to `UpsertItemAsync()` but for Patch operations.

Currently, I do a `PatchItemAsync()` and if document does not exist it throws a 404 Not Found exception and then I do a `CreateItemAsync()` for that item.

Problem is I want to do a Bulk Insert using `AllowBulkExecution = true` and this is not a good approach.

My other idea is to fetch all the document id's and conditionally create a `List<Task>` (for bulk) group for `PatchItemAsync()` and `CreateItemAsync()`. Not really happy with this but I don't see another way.

Question, is there a better solution to handle such cases using SDK?

**Comment by cpirtea-bun** [comment]
I've been waiting for this feature for a long time. MongoDb had this 5 years ago and it was exceptionally useful when some properties were always safe to override.
